### PR TITLE
PHP CS Fixer configuration generation in constructed project

### PR DESCRIPTION
**Pull request by raphaelstolt**

### CODE REVIEW

**Comment by jonathantorres** [comment]
I liked `php-cs-fixer`. But I don't know if everyone else will actually use it on every project. So, I would add this as an option. Something like `--phpcs` Just like it's done with the `--git` option. 

What do you think @hannesvdvreken ?

**Comment by hannesvdvreken** [comment]
Didn't follow this entirely, what does the --git option do? Run git init?

Next to .phpcs there is also scrutinizer and styleci to validate code style.

**Comment by raphaelstolt** [comment]
@hannesvdvreken Yes it does a `git init` on the constructed project.

Will look into `scrutinizer` and ~~`circleci`~~ `styleci` and prolly make them values for ther `--phpcs` option.  Not 100% sure about  scrutinizer as it might only be  free for OSS projects.

**Comment by hannesvdvreken** [comment]
Don't forget about styleci ;-)

**Comment by raphaelstolt** [comment]
Ooops sorry,  meant styleci (but apparently there's also a circleci CI platform).

**Comment by raphaelstolt** [comment]
@jonathantorres What's your POV on adding scrutinizer and styleci?

**Comment by hannesvdvreken** [comment]
My 2c, if you're doing `.php_cs`: default to PSR-2 level (most common) and optionally let the user set a different level (Symfony or PSR-1), but not more than that, because there are too many options and too many damn opinions.

File: src/stubs/gitattributes.txt [hunk 1]
```diff
@@ -4,6 +4,7 @@
 /.gitattributes export-ignore
 /.gitignore export-ignore
 /.travis.yml export-ignore
+/.php_cs export-ignore
```
Comment (hannesvdvreken) [inline src/stubs/gitattributes.txt:7]: y u no move it up one line? makes it ordered. just me nitpicking, but...

File: src/stubs/phpcs.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+$finder = Symfony\CS\Finder\DefaultFinder::create()
+    ->in(__DIR__);
+
+return Symfony\CS\Config\Config::create()
+    ->fixers(array())
```
Comment (hannesvdvreken) [inline src/stubs/phpcs.txt:7]: wow, never seen that array syntax before... :trollface:

File: src/stubs/gitattributes.txt (hunk 1 above)
Comment (raphaelstolt) [inline src/stubs/gitattributes.txt:7]: I guess this needs to be refactored towards using the `exportIgnores` array of the Construct class anyways, but will keep that in mind when doing so ;D

**Comment by hannesvdvreken** [comment]
@raphaelstolt @jonathantorres finish this PR for adding the `.php_cs` file only. It's optional so people can contribute `--scrutinizer` flag if they want to. So eventually people can have both `.php_cs` and `.scrutinizer.yml` if they want to.

File: src/stubs/gitattributes.txt (hunk 1 above)
Comment (hannesvdvreken) [inline src/stubs/gitattributes.txt:7]: :+1:

**Comment by jonathantorres** [comment]
Yeah. Let's make the php-cs file configuration optional. Let's stick with `psr-2` and if the user doesn't provide the `--phpcs` option simply don't generate the file.

I'm totally open to doing the same with scrutinizer and styleci. Adding the `--scrutinizer` and `--styleci` options.

**Pull request closed without merging**